### PR TITLE
[DASH-2365] Add pagination for Data Objects

### PR DESCRIPTION
**Pull request by wpasinski**
zrzut ekranu 2016-11-28 o 20 40 02
zrzut ekranu 2016-11-28 o 20 40 12
zrzut ekranu 2016-11-28 o 20 40 21

### CODE REVIEW

File: src/apps/DataObjects/DataObjects.jsx [hunk 1]
```diff
@@ -93,35 +89,42 @@ const DataObjects = React.createClass({
...
-        isLoading={isLoading}
-        items={items}
-        users={users}
-        hasNextPage={hasNextPage}
-        selectedRows={selectedRows}
-        classObject={classObj}
-        handleRowSelection={this.handleRowSelection}
-        handleMoreRows={this.handleMoreRows}
-      />
+      <Loading show={isLoading}>
```
Comment (patrykkopycinski) [inline src/apps/DataObjects/DataObjects.jsx]: why did you move loading here?

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
+import React, { Component } from 'react';
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: why it's in common as it's `DataObjects` specific `Table`?

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      columns: this.getInitialColumns()
+    };
+  }
+
+  getStyles = () => ({
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: it doesn't have to be a method

File: src/common/TableHeaderSortableColumn/TableHeaderSortableColumn.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  const renderSortingIcon = () => {
+    if (!sortable) {
+      return null;
+    }
+
+    let className = null;
+
+    if (currentOrderBy === id) className = 'synicon-sort-ascending';
```
Comment (patrykkopycinski) [inline src/common/TableHeaderSortableColumn/TableHeaderSortableColumn.js]: use `classnames` to handle conditions for `className`

File: src/common/TableHeaderSortableColumn/TableHeaderSortableColumn.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    return (
+      <FontIcon
+        className={className}
+        style={styles.fontIcon}
+      />
+    );
+  };
+
+  const renderChildren = () => {
+    if (sortable) {
```
Comment (patrykkopycinski) [inline src/common/TableHeaderSortableColumn/TableHeaderSortableColumn.js]: `if (!sortable) { return children }`

File: src/common/TableHeaderSortableColumn/TableHeaderSortableColumn.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
```
Comment (patrykkopycinski) [inline src/common/TableHeaderSortableColumn/TableHeaderSortableColumn.js]: remove it

File: src/common/Pagination/Pagination.sass [hunk 7]
```diff
@@ -0,0 +1,36 @@
+.pagination
```
Comment (patrykkopycinski) [inline src/common/Pagination/Pagination.sass]: we're going to remove .sass completely, so please rewrite it to JS module

File: src/common/Pagination/Pagination.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import ReactPaginate from 'react-paginate';
+
+import './Pagination.sass';
+
+const Pagination = ({ pageNum, currentPage, clickCallback }) => {
+  const handleClick = ({ selected }) => {
+    clickCallback(selected + 1);
```
Comment (patrykkopycinski) [inline src/common/Pagination/Pagination.js]: use react convention names, it should be called `onClick`

File: src/common/DataObjectsTable/DataObjectsTableCell.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    reference: (value) => renderReference(value),
+    relation: (value) => renderReference(value)
+  };
+
+  const renderContent = (content) => {
+    if (!content) {
+      return null;
+    }
+
+    let renderedContent = content;
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTableCell.js]: instead you should just `return` in proper place instead assigning it to temporary variable

File: src/common/DataObjectsTable/DataObjectsTableCell.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  const renderReference = (value) => (
+    <div>{`${value.target}: ${value.value}`}</div>
+  );
+
+  const objectTypeRenderMethods = {
+    datetime: (value) => renderDate(value.value),
+    file: (value) => renderFile(value),
+    reference: (value) => renderReference(value),
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTableCell.js]: ` reference: renderReference`

File: src/common/DataObjectsTable/DataObjectsTableCell.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import _ from 'lodash';
+import Moment from 'moment';
+
+import { FontIcon } from 'material-ui';
+
+const DataObjectsTableCell = ({ item, columnId, users }) => {
+  const renderDate = (value) => {
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTableCell.js]: each of type should be a separate component

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+      });
+    }
+
+    return columns;
+  }
+
+  getCheckedColumnsIDs() {
+    const { columns } = this.state;
+
+    return _.map(columns, (column) => {
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: i think you should remove `null`s  from the array before you `return` them

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+  updateLocalStorage() {
+    const { className } = this.props.classObject;
+
+    localStorage.setItem(`dataobjects_checkedcolumns_${className}`, JSON.stringify(this.getCheckedColumnsIDs()));
+  }
+
+  checkToggleColumn(columnId) {
+    const { columns } = this.state;
+
+    columns.forEach((item) => {
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: it's a bad idea to modify current state

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+    this.updateLocalStorage();
+    this.setState({ columns });
+  }
+
+  renderTableHeader() {
+    const styles = this.getStyles();
+    const { columns } = this.state;
+    const { currentOrderBy, handleSortingSelection } = this.props;
+
+    const columnsComponents = _.map(columns, (item, index) => {
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: the same here, remove `null`s before you `return` them

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 15]
```diff
@@ -0,0 +1,245 @@
...
+          key={`header-column-${index}`}
+          style={{
+            width: item.width || 100,
+            whiteSpace: 'normal',
+            wordWrap: 'normal'
+          }}
+          id={item.id}
+          sortable={item.sortable}
+          currentOrderBy={currentOrderBy}
+          clickHandler={handleTableHeaderSortableColumnClick}
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: `clickHandler` => `onClick`

File: src/common/DataObjectsTable/DataObjectsTable.js (hunk 14 above)
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: callback should be a separate component

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 16]
```diff
@@ -0,0 +1,245 @@
...
+      <TableHeader key="header">
+        <TableRow key="header-row">
+          <TableHeaderColumn
+            key="header-column-check-menu"
+            tooltip="Manage columns"
+            style={styles.tableHeaderColumn}
+          >
+            <ColumnsFilterMenu
+              columns={columns}
+              checkToggleColumn={(columnId) => this.checkToggleColumn(columnId)}
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: `checkToggleColumn={this.checkToggleColumn}`

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+      </TableHeader>
+    );
+  }
+
+  renderTableData() {
+    const styles = this.getStyles();
+    const { columns } = this.state;
+    const { users, items, selectedRows } = this.props;
+
+    const tableData = _.map(items, (item, index) => {
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: move callback to a separate component

File: src/common/DataObjectsTable/DataObjectsTable.js [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+      const selected = (selectedRows || []).indexOf(index) > -1;
+      const columnsComponents = _.map(columns, (column, idx) => {
```
Comment (patrykkopycinski) [inline src/common/DataObjectsTable/DataObjectsTable.js]: the same here

File: src/common/Pagination/Pagination.sass (hunk 7 above)
Comment (wpasinski) [inline src/common/Pagination/Pagination.sass]: it's an external component. i'm not able to style it via props.

File: src/apps/DataObjects/DataObjects.jsx (hunk 1 above)
Comment (wpasinski) [inline src/apps/DataObjects/DataObjects.jsx]: because I want to hide pagination on loading too, to prevent it from jumping.
